dev/ts/component/SvgExporter.ts

```diff
@@ -0,0 +1,202 @@
+// https://github.com/jgraph/drawio/blob/v14.7.7/src/main/webapp/js/diagramly/Editor.js#L5932
+// https://github.com/jgraph/drawio/blob/v14.8.0/src/main/webapp/js/grapheditor/Graph.js#L9007
+export class SvgExporter {
+  constructor(private graph: mxGraph) {}
+
+  public exportSvg(): string {
+    return this.doSvgExport(true);
+  }
+
+  public exportSvgForPng(): string {
+    // chrome and webkit: tainted canvas when svg contains foreignObject
+    // also on brave --> probably fail on chromium based browsers
+    // so disable foreign objects for such browsers
+    const isFirefox = mxgraph.mxClient.IS_FF;
+    return this.doSvgExport(isFirefox);
+  }
+
+  private doSvgExport(enableForeignObjectForLabel: boolean): string {
+    const svgDocument = this.computeSvg({ scale: 1, border: 25, enableForeignObjectForLabel: enableForeignObjectForLabel });
+    const svgAsString = mxgraph.mxUtils.getXml(svgDocument);
+    return `<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
+${svgAsString}
+`;
+  }
+
+  private computeSvg(svgExportOptions: SvgExportOptions): XMLDocument {
+    const scale = svgExportOptions.scale ?? 1;
+    const border = svgExportOptions.border ?? 0;
+    const crisp = svgExportOptions.crisp ?? true;
+    const enableForeignObjectForLabel = svgExportOptions.enableForeignObjectForLabel ?? true;
+
+    const bounds = this.graph.getGraphBounds();
+    const viewScale = this.graph.view.scale;
+
+    // Prepares SVG document that holds the output
+    const svgDoc = mxgraph.mxUtils.createXmlDocument();
+    const root = svgDoc.createElementNS(mxgraph.mxConstants.NS_SVG, 'svg');
+
+    const s = scale / viewScale;
+    const w = Math.max(1, Math.ceil(bounds.width * s) + 2 * border);
+    const h = Math.max(1, Math.ceil(bounds.height * s) + 2 * border);
+
+    root.setAttribute('version', '1.1');
+    root.setAttribute('width', w + 'px');
+    root.setAttribute('height', h + 'px');
+    root.setAttribute('viewBox', (crisp ? '-0.5 -0.5' : '0 0') + ' ' + w + ' ' + h);
+    svgDoc.appendChild(root);
+
+    const group = svgDoc.createElementNS(mxgraph.mxConstants.NS_SVG, 'g');
+    root.appendChild(group);
+
+    const svgCanvas = this.createSvgCanvas(group);
+    svgCanvas.foEnabled = enableForeignObjectForLabel;
+    // Comments from draw.io
+    // Renders graph. Offset will be multiplied with state's scale when painting state.
+    // TextOffset only seems to affect FF output but used everywhere for consistency.
+    svgCanvas.foOffset = crisp ? -0.5 : 0;
+    svgCanvas.textOffset = crisp ? -0.5 : 0;
+    svgCanvas.imageOffset = crisp ? -0.5 : 0;
+    svgCanvas.translate(Math.floor((border / scale - bounds.x) / viewScale), Math.floor((border / scale - bounds.y) / viewScale));
+
+    svgCanvas.scale(s);
+
+    const imgExport = new mxgraph.mxImageExport();
+    // FIXME only the first overlay is placed at the right position
```

Comment:
To do later ?

Reply:
I think so, we fully rely on the mxgraph code for overlay export. We may have introduce side effects with our custom overlays management.
I didn't find the root cause of the issue, and as I don't want to spend more time on this topic, I suggest we defer it.